tests/test_pump.py

```diff
@@ -26,8 +26,8 @@ def test_pump_4_3_liters(self):
         # verify that the values are equal to several significant digits.
         self.assertAlmostEqual(wait_time_expected, wait_time_actual)
         # Verify the signal gets sent to turn the pump pin on, then off.
-        self.mock_pi_io.turn_pin_on.assert_called_once_with(pi_io.PIN_PUMP)
-        self.mock_pi_io.turn_pin_off.assert_called_once_with(pi_io.PIN_PUMP)
+        self.mock_pi_io.turn_pin_on.assert_called_once_with(self.pump_pin)
```

Comment:
Ideally the reader shouldn't have to read the `setUp` function to understand why a test passes. For things like `self.mock_clock` I feel like it's easy for the reader to assume it's the pump's internal clock. For things like the GPIO pin number where it's set to a particular value, I feel like it's less easy to assume and the reader has to go to the `setUp` to understand what's going on.

Suggest instantiating the pump in each test function like:

```python
pump.Pump(self.mock_pi_io, self.mock_clock, pump_pin=5).pump_water(4300.0)
...
self.mock_pi_io.turn_pin_on.assert_called_once_with(5)
self.mock_pi_io.turn_pin_off.assert_called_once_with(5)
```

That way, the reason for the pin being a particular value is obvious within the test function itself.